Updating docs for unify mac changes (#46605)
gravitational/teleport: CHANGELOG.md

## 17.0.0 (11/xx/2024)

### Breaking changes

> #### Mac Package Changes

Reviewer: ```suggestion
#### macOS Package Changes
```

> To support a more consistent user experience, Mac packages have been restructured so that all packages will provide a 

Reviewer: ```suggestion
To support a more consistent user experience, the macOS packages have been restructured so that all packages will provide a 
```

> `tsh` and `tctl` capable of utilizing TouchID

Reviewer: ```suggestion
`tsh` and `tctl` capable of utilizing TouchID.
```

> All Mac packages now include `tsh.app` and `tctl.app` bundles and the standalone binaries for `tsh` and `tctl` have been removed. 

Reviewer: ```suggestion
The standalone `tsh` and `tctl` binaries have been removed from these packages and
replaced with `tsh.app` and `tctl.app` bundles.
```
